### ISSUE TITLE
Null check operator error when using CTE in drift file

### ISSUE THREAD

**Issue by austinried** [simolus3/moor#2397]
I'm trying to use this query with a CTE in my drift file:

```sql
shuffleQueue:
  WITH shuffle_order AS (
    SELECT ROWID, row_number() OVER (ORDER BY RANDOM()) rn
    FROM queue
  )
  UPDATE queue SET
    shuffle_index = (SELECT rn FROM shuffle_order WHERE shuffle_order.ROWID = queue.ROWID) - 1
  FROM shuffle_order;
```

But I'm getting this error during build:

```
[WARNING] drift_dev on lib/database/tables.drift:
Could not analyze DriftElementId(package:subtracks2/database/tables.drift, shuffleQueue)
Null check operator used on a null value
```

The query otherwise works fine when I test it against the database.  Here is the table schema if it helps:

```sql
CREATE TABLE queue(
  "index" INT NOT NULL PRIMARY KEY,
  source_id INT NOT NULL,
  id TEXT NOT NULL,
  context ENUM(QueueContextType) NOT NULL,
  context_id TEXT,
  current_track BOOLEAN UNIQUE,
  shuffle_index INT
);
```

**Comment by simolus3** [comment]
Interesting, our sql analyzer runs into two separate issues with that query. First, we resolve CTEs too late for `UPDATE` statements, which causes the crash (fixed in 611fd4c4e95712077a7aba3dbcc80c8b4991c2dc).

Then there's another weird edge case when `ROWID` references are used to a table that has a rowid alias (e.g. `id INTEGER NOT NULL PRIMARY KEY`). Outside of CTEs and subqueries, the column takes the name of the alias. But in those clauses, it doesn't. I didn't know that was a thing, so there's actually another bug in sqlparser causing the `shuffle_order.ROWID` reference to not work. That has been fixed in db15b792ca2ad9394975cc5a026e06758a03980f.

I thought I had sqlite3 figured out by now but these edge cases just keep coming up :D So, thanks a lot for the report!

**Comment by austinried** [comment]
Just trying to keep you on your toes :wink: 

Thanks again as usual!
